experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,207 @@
+  Modules=[multipartition([lbb...]) for lbb in class_parameters(charTable)]
+  l=length(Modules)
+  charTirr=reorder(charTirr,r,n,Modules,K,mps)
+  charTirrT=solve_init(matrix(K,transpose(charTirr)))
+  rows=zero_matrix(Q,l,0)
+  for i in 1:l
+    col=zero_matrix(Q,l,1)
+    for j in 1:l
+      v=matrix(K,l,1,map(k->charTirr[i,k]*charTirr[j,k],1:l))
+      x=solve(charTirrT,v,side=:right)
+      col=col+fake_deg(mps[j],K,Q,var)*x
+    end
+    rows=hcat(rows,col)
+  end
+  return rows
+end
+
+# computes the character of the simple representations of the Cherednik algebra.
+function C_L(r::Int, n::Int, K::QQAbField{AbsSimpleNumField}, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}}, var::AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}, mps::Vector{Multipartition{Int}})
+  C_D=C_Delta(r,n,K,Q,var,mps)
+  d = [var^b_inv(mp)//fake_deg(mp,K,Q,var) for mp in mps]
+  diag=diagonal_matrix(Q,d)
+  return diag * C_D
+end
+
+function bigger_ord(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int}, mps::Vector{Multipartition{Int}})
+  r=length(lbb)
+  n=sum(lbb)
+  lbbquot=tau_om(lbb,wperm,coroot)
+  res=map(x->x,Iterators.filter(x-> dominates(tau_om(x,wperm,coroot),lbbquot),mps))
+  return res
+end
+
+function smaller_ord(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int}, mps::Vector{Multipartition{Int}})
+  r=length(lbb)
+  n=sum(lbb)
+  lbbquot=tau_om(lbb,wperm,coroot)
+  res=map(x->x,Iterators.filter(x-> dominates(lbbquot,tau_om(x,wperm,coroot)),mps))
+  return res
+end
+
+function wreath_macs(n::Int, r::Int, wperm::PermGroupElem, coroot::Vector{Int})
+  K,_ = abelian_closure(QQ)
+  R, (q,t) = polynomial_ring(K, [:q,:t])
+  Q = fraction_field(R)
+
+  mps=collect(multipartitions(n,r))
+  l=length(mps)
+
+  c_L=C_L(r,n,K,Q,t,mps)
+  c_L_q=map_entries(f->numerator(f)(0,q)//denominator(f)(0,q),c_L)
+  c_L_tinv=map_entries(f->numerator(f)(0,1//t)//denominator(f)(0,1//t),c_L)
+
+  rows=[]
+  for lbb in mps
+    smallers=smaller_ord(lbb, wperm, coroot,mps) #tinv
+    biggers=bigger_ord(lbb, wperm, coroot,mps) #q
+    smaller_indices=sort!(indexin(smallers,mps))
+    bigger_indices=sort!(indexin(biggers,mps))
+    sub_smaller_tinv=vcat(map(i->c_L_tinv[i:i,:],smaller_indices)...)
+    sub_bigger_q=vcat(map(i->c_L_q[i:i,:],bigger_indices)...)
+    M=vcat(sub_smaller_tinv,sub_bigger_q)
+    B=kernel(M, side=:left)
+    rows=vcat(rows,B[1,1:length(smaller_indices)]*sub_smaller_tinv)
+  end
+  c_L_qt = matrix(Q,l,l,rows)
+
+  triv=[partition([n])]
+  for i in 2:r
+    push!(triv,partition([]))
+  end
+  triv=multipartition(triv)
+  index_triv=findfirst(x-> x==triv,mps)
+  d=[1//c_L_qt[i,index_triv] for i in 1:l]
+  diag=diagonal_matrix(Q,d)
+  c_L_qt_H=diag*c_L_qt
+  return c_L_qt_H
+end
+
+function wreath_mac(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int})
+  K,_ = abelian_closure(QQ)
+  R, (q,t) = polynomial_ring(K, [:q,:t])
```

Comment:
```suggestion
  R, (q,t) = polynomial_ring(K, [:q,:t]; cached=false)
```
please have a look at https://docs.oscar-system.org/dev/DeveloperDocumentation/caching/ on how to instead pass an appropriate parent for this function

Reply:
I will add that.

Reply:
For now I have disabled caching of the polynomial_ring  in both `wreath_macs` and `wreath_mac`. The issue I have is now that when I want to compare results in my tests, polynomials will never be the same since I have to recreate the polynomial ring in the tests. A solution would be to require the ring as an argument of the two functions `wreath_macs` and `wreath_mac`. What is your opinion on that?

Reply:
yes passing the parent ring as an (optional) argument is how we usually handle this.

(Side note: I thought we'd explain this, including examples, on https://docs.oscar-system.org/dev/DeveloperDocumentation/caching/ but it seems we don't.

Turns out this is under https://docs.oscar-system.org/dev/DeveloperDocumentation/styleguide/#Optional-arguments-for-parents-of-return-values ! Ugh.